pkgs/build-support/rust/default.nix

```diff
@@ -75,8 +80,17 @@ let
   cxxForBuild="${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}c++";
   ccForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
   cxxForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}c++";
-  releaseDir = "target/${rustTarget}/${buildType}";
-  tmpDir = "${releaseDir}-tmp";
+
+  releaseDirPrefix = if impureCacheDir == null
+    then "target"
+    else
+      assert
+        stdenv.lib.assertMsg (stdenv.lib.hasPrefix "/" impureCacheDir)
+        "impureCacheDir must be an absolute path.";
+      impureCacheDir;
```

Comment:
Maybe `impureCacheDir` could just be renamed `cacheDir` with the default set to `"target"`? Then this absolute path check is not necessary.

I think this would be a good compromise, because then the derivation does not encourage impure use, but anyone who is willing to disable the Nix sandbox could still set it to an absolute directory. I guess setting `CARGO_TARGET_DIR` to `target` as a result does not have a negative impact, except for causing a lot of rebuilds ;).

Reply:
> Maybe impureCacheDir could just be renamed cacheDir with the default set
 to "target"? Then this absolute path check is not necessary.

That could be done.

>  ...because then the derivation does not encourage impure use
I'm not sure. This just sounds like hiding the true purpose because I can't
think another reason to set the cache directory.

One possible alternative would be allow passing in the cache directory
manually and provide the resulting cache directory as a separate output.
This is weirder to use and technically pure. However I decided against this
solution as it didn't seem to add any real value other than working without
sandboxing.